src/pool/srv_pool.c

```diff
@@ -447,6 +447,87 @@ pool_prop_write(struct rdb_tx *tx, const rdb_path_t *kvs, daos_prop_t *prop)
 	return rc;
 }
 
+static int
+pool_buf_fill_extend(struct pool_map *map, struct pool_buf *map_buf,
+		int nnodes, int map_version, int ndomains,
+		const int32_t *domains, uuid_t target_uuids[],
+		const d_rank_list_t *target_addrs, uuid_t *uuids, bool *updated)
+{
+	struct pool_component	map_comp;
+	uint32_t		num_comps;
+	struct pool_domain	*found_dom;
+	int i, rc;
+
+	num_comps = pool_map_find_domain(map, PO_COMP_TP_RACK, PO_COMP_ID_ALL,
+					 NULL);
+	/* fill domains */
+	for (i = 0; i < ndomains; i++) {
+		map_comp.co_type = PO_COMP_TP_RACK;	/* TODO */
+		map_comp.co_status = PO_COMP_ST_NEW;
+		map_comp.co_index = i;
```

Comment:
co_index should add num_comps as well? in case this will be used later.

Reply:
Fixed